### PR TITLE
Added support for URI(Uniform Resource Identifier) records.

### PR DESCRIPTION
**Pull request by martin-256**
I'm finishing work from pull request https://github.com/c-ares/c-ares/pull/389 created by my former colleague https://github.com/Prasek-prtes.

Added tests and forgotten adjustments.

### CODE REVIEW

File: src/lib/ares_parse_uri_reply.c [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+#endif
+#ifdef HAVE_ARPA_NAMESER_H
+#  include <arpa/nameser.h>
+#else
+#  include "ares_nameser.h"
+#endif
+#ifdef HAVE_ARPA_NAMESER_COMPAT_H
+#  include <arpa/nameser_compat.h>
+#endif
+
```
Comment (bradh352) [inline src/lib/ares_parse_uri_reply.c:37]: Please just include ares_nameser.h always, and never directly include arpa/nameser.h or arpa/nameser_compat.h ... see c2f3235 on this change we made to the rest of c-ares

**Review by bradh352 — COMMENTED** [review]
at a cursory glance, the only issue I see is the headers mentioned in the review

File: src/lib/ares_parse_uri_reply.c (hunk 1 above)
Comment (martin-256) [reply src/lib/ares_parse_uri_reply.c:31]: Thank you. I didn't notice this change.

**Comment by bradh352** [comment]
I'll merge this after the 1.17.2 release which is just about out.

**Pull request merged**